lib/idempo.rb

```diff
@@ -83,8 +85,10 @@ def call(env)
 
   private
 
-  def rack_v3?
-    Gem::Version.new(Rack.release) >= Gem::Version.new("3.0")
+  def downcase_keys(raw_headers)
+    raw_headers.each_with_object({}) do |(name, value), hh|
+      hh[name.to_s.downcase] = value
```

Comment:
**suggestion:** we're allocating another hash here from headers, but keys will be allocated in memory additionally _on every request_. If we use down cased symbols as keys, this should reduce additional allocation to absolute minimum.

Plenty of keys repeat, so I assume symbols should work really nice here.

Reply:
Symbols can't be used as it would make the header hash out of Rack spec. Allocating an extra hash per request seems fine to be honest